File: CPP/Algorithm/Is_Substring.cpp
// Complexity: O(|main| * |sub|)
// STL Approach:

bool sstr(auto &main, auto &sub) {
    return (main.find(sub) != -1);
}

// Iterative Approach:

bool sstr(auto &main, auto &sub) {
    int m = main.size(), n = sub.size();
    for (int i = 0, j; i <= m - n; ++i) {
        for (int j = 0; j < n; ++j) {
            if (main[i + j] != sub[j]) break;
		}
        if (j == n) return true;
    }
    return false;
}

File: Python/Algorithm/Is_Substring.py
# Complexity: O(|main| * |sub|)

# Built-in Approach:

def sstr(main, sub): return sub in main

# Iterative Approach:

def sstr(main, sub):
    m, n = len(main), len(sub)
    for i in range(m - n + 1):
        for j in range(n):
            if main[i + j] != sub[j]: break
        else: return True
    return False